### PR TITLE
Couple of hosting improvements

### PR DESCRIPTION
**Pull request by mkangia**
- Share status
- Allow retry/recreate

[Ticket](https://dimagi-dev.atlassian.net/browse/ICDS-728)

FF: `MANAGE_CCZ_HOSTING`
fyi @esoergel

### CODE REVIEW

File: custom/icds/static/icds/js/manage_hosted_ccz.js [hunk 1]
```diff
@@ -21,6 +21,10 @@ hqDefine('icds/js/manage_hosted_ccz', [
             self.fileName = fileName;
             self.note = note;
             self.status = status;
+            self.rowColor = function () {
```
Comment (mkangia) [inline custom/icds/static/icds/js/manage_hosted_ccz.js]: my original attempt was to use `class` but i just could not get it working with knockout js so i then fell back to `color`.
I had rowClass as a computed property which returned `info`, 'danger` etc and then i did `data-bind=class: rowClass` on `tr` but the class never came into action.
I will be happy to revert this change to class.

Comment (orangejenny) [reply custom/icds/static/icds/js/manage_hosted_ccz.js]: `class` was added in 3.5 and we're only on 3.3: https://github.com/knockout/knockout/releases

Could you use `css`? https://knockoutjs.com/documentation/css-binding.html

I also wouldn't object to upgrading to 3.5.

Comment (mkangia) [reply custom/icds/static/icds/js/manage_hosted_ccz.js]: Updated to use `css`. 
Upgrade would need a robust testing.

File: custom/icds/forms.py [hunk 2]
```diff
@@ -54,6 +54,16 @@ class HostedCCZForm(forms.Form):
                                  required=False, widget=Select(choices=[]))
     file_name = forms.CharField(label=ugettext_lazy("CCZ File Name"), required=False)
     note = forms.CharField(required=False, widget=forms.Textarea(attrs={'rows': 3, 'cols': 15}))
+    status = forms.ChoiceField(label=ugettext_lazy("Status"),
+                               choices=(
+                                   ('', 'Select Status'),
```
Comment (orangejenny) [inline custom/icds/forms.py]: Not blocking, but can you use an actual label or placeholder for this? It always strikes me as weird to see a placeholder as an actual piece of data.

File: custom/icds/models.py [hunk 3]
```diff
@@ -92,6 +92,12 @@ class HostedCCZ(models.Model):
     profile_id = models.CharField(max_length=255, blank=True)
     file_name = models.CharField(max_length=255, blank=True)
     note = models.TextField(blank=True)
+    status = models.CharField(max_length=255, null=False, blank=False, default='pending',
+                              choices=(('pending', 'Pending'),
```
Comment (orangejenny) [inline custom/icds/models.py]: Nitpick: constants are better than magic strings, both here and in `update_status` below.

File: custom/icds/templates/icds/manage_hosted_ccz.html [hunk 4]
```diff
@@ -81,7 +81,7 @@ <h1>
               <td data-bind="text: note">
               </td>
               <td>
-                <a data-bind="attr: { href: url}">
+                <a data-bind="attr: { href: removeUrl}" class="btn btn-danger">
```
Comment (orangejenny) [inline custom/icds/templates/icds/manage_hosted_ccz.html]: hooray, buttons are definitely better than links for actions

File: custom/icds/templates/icds/manage_hosted_ccz.html [hunk 5]
```diff
@@ -84,6 +85,9 @@ <h1>
                 <a data-bind="attr: { href: removeUrl}" class="btn btn-danger">
                   {% trans 'Remove' %}
                 </a>
+                <a data-bind="attr: { href: recreateUrl}" class="btn btn-warning">
```
Comment (orangejenny) [inline custom/icds/templates/icds/manage_hosted_ccz.html]: `btn-default`, please, we don't use `btn-warning`

File: custom/icds/templates/icds/manage_hosted_ccz.html [hunk 6]
```diff
@@ -64,7 +64,8 @@ <h1>
             </tr>
           </thead>
           <tbody data-bind="foreach: hostings">
-            <tr data-bind="style: {background: rowColor()}">
+            <tr data-bind="css: {'success': status === 'completed', 'info': status === 'pending',
```
Comment (orangejenny) [inline custom/icds/templates/icds/manage_hosted_ccz.html]: so much cleaner!

**Review by orangejenny — APPROVED** [review]

File: custom/icds/templates/icds/manage_hosted_ccz.html (hunk 5 above)
Comment (mkangia) [reply custom/icds/templates/icds/manage_hosted_ccz.html]: umm, i am using that for the colour it provides along with the style.
Why not use `btn-warning`?

Comment (orangejenny) [reply custom/icds/templates/icds/manage_hosted_ccz.html]: We don't use it anywhere else in HQ. See https://www.commcarehq.org/styleguide/molecules/#molecules-buttons

File: custom/icds/models.py [hunk 7]
```diff
@@ -163,7 +168,7 @@ def delete(self, *args, **kwargs):
         super(HostedCCZ, self).delete(*args, **kwargs)
 
     def update_status(self, new_status):
-        assert new_status in ['pending', 'building', 'failed', 'completed']
+        assert new_status in self.STATUSES
```
Comment (orangejenny) [inline custom/icds/models.py]: nice, thank you

File: custom/icds/forms.py (hunk 2 above)
Comment (mkangia) [reply custom/icds/forms.py]: Can you share an example of how that would be?

File: custom/icds/templates/icds/manage_hosted_ccz.html (hunk 5 above)
Comment (mkangia) [reply custom/icds/templates/icds/manage_hosted_ccz.html]: Okay, updated. Kept both as dangerous.

File: custom/icds/forms.py (hunk 2 above)
Comment (mkangia) [reply custom/icds/forms.py]: Offlined, fine to use the blank option. Without it, the first option is pre-selected.

**Review by orangejenny — APPROVED** [review]

**Pull request merged**